src/transport.js

```diff
@@ -1,26 +1,99 @@
-import request from 'request-promise'
-import promiseRetry from 'promise-retry'
+import fetch from 'cross-fetch'
+// from https://github.com/jonbern/fetch-retry
+// and https://github.com/jonbern/fetch-retry/pull/27
+export default (url, options) => {
+  var retries = 3
+  var retryDelay = 1000
+  var retryOn = [429, 502, 503, 504]
+  if (options && options.retries !== undefined) {
+    if (isPositiveInteger(options.retries)) {
+      retries = options.retries
+    } else {
+      throw new ArgumentError('retries must be a positive integer')
+    }
+  }
 
-// When the API sends a 429 back, retry it for 10 times
-// We also retry on server errors 502, 503 and 504
-// This code is mainly copied from
-// https://github.com/CanTireInnovations/request-promise-retry/blob/master/index.js
+  if (options && options.retryDelay !== undefined) {
+    if (
+      isPositiveInteger(options.retryDelay) ||
+      typeof options.retryDelay === 'function'
+    ) {
+      retryDelay = options.retryDelay
+    } else {
+      throw new ArgumentError(
+        'retryDelay must be a positive integer or a function returning a positive integer'
+      )
+    }
+  }
 
-const retryRequest = promiseRequest => (req, options) =>
-  promiseRetry(
-    retry =>
-      promiseRequest(req).catch(err => {
-        if (
-          err.statusCode !== 429 &&
-          err.statusCode !== 502 &&
-          err.statusCode !== 503 &&
-          err.statusCode !== 504
-        ) {
-          throw err
-        }
-        retry(err)
-      }),
-    options
-  )
+  if (options && options.retryOn) {
+    if (
+      Array.isArray(options.retryOn) ||
+      typeof options.retryOn === 'function'
+    ) {
+      retryOn = options.retryOn
+    } else {
+      throw new ArgumentError('retryOn property expects an array or function')
+    }
+  }
 
-export default retryRequest(request)
+  return new Promise(function (resolve, reject) {
+    var wrappedFetch = function (attempt) {
+      fetch(url, options)
+        .then(function (response) {
+          if (
+            Array.isArray(retryOn) &&
+            retryOn.indexOf(response.status) === -1
+          ) {
+            resolve(response)
+          } else if (typeof retryOn === 'function') {
+            if (retryOn(attempt, null, response)) {
+              retry(attempt, null, response)
+            } else {
+              resolve(response)
+            }
+          } else {
+            if (attempt < retries) {
+              retry(attempt, null, response)
+            } else {
+              resolve(response)
+            }
+          }
```

Comment:
I'd simplify above bit by using `return` instead of `else`. Less indentation -> easier to read.
Also maybe we could use async/await? Unsure if that's supported in our build setup tho.

Reply:
simplified. we could use async/await (I use it now also in other places), but prefer to leave it with this working code for now